src/lib/server/tools/utils.ts

```diff
@@ -16,9 +16,21 @@ type GradioResponse = {
 export async function callSpace<TInput extends unknown[], TOutput extends unknown[]>(
 	name: string,
 	func: string,
-	parameters: TInput
+	parameters: TInput,
+	ipToken: string | undefined
 ): Promise<TOutput> {
-	const client = await Client.connect(name, {
+	class CustomClient extends Client {
```

Comment:
implementation-wise, it was easier/simpler to define  `CustomClient` inside `callSpace` because `GradioClient.connect` is a static method